index.html

```diff
@@ -1142,67 +1142,89 @@
 
 <!-- - - - - - - - - - - - - Data representation - - - - - - - - - - - - - -->
 <section> <h2>Data Representation</h2>
-  <section> <h3>The <dfn>NDEFMessage</dfn> dictionary</h3>
+  <section> <h3>The <dfn>NDEFMessage</dfn> interface</h3>
     <p>
       The content of any <a>Web NFC message</a> is exposed by the
-      <a>NDEFMessage</a> dictionary:
+      <a>NDEFMessage</a> interface:
     </p>
     <pre class="idl">
-      dictionary NDEFMessage {
+      [Constructor(NDEFMessageInit messageInit), Exposed=Window]
+      interface NDEFMessage {
+        readonly attribute USVString url;
+        readonly attribute FrozenArray&lt;NDEFRecord&gt; records;
+      };
+
+      dictionary NDEFMessageInit {
         USVString url;
-        sequence&lt;NDEFRecord&gt; records;
+        sequence&lt;NDEFRecordInit&gt; records;
       };
     </pre>
     <p>
       The <dfn>NDEFMessage.url</dfn>
       property represents the <a>Web NFC Id</a> of a received
-      <a>Web NFC message</a>. When used in the <code><a>NFCWriter.push()</a></code>
-      method, it represents a <a>URL path</a> used for constructing the
-      <a>Web NFC Id</a> of the pushed <a>Web NFC content</a>.
+      <a>Web NFC message</a>.
     </p>
     <p>
       The <dfn>NDEFMessage.records</dfn>
       property represents a <a>list</a> of <a>NDEF record</a>s defining the
       <a>Web NFC message</a>.
     </p>
+    <p>
+      The <dfn>NDEFMessageInit</dfn> dictionary is used to initialize a
+      <a>Web NFC message</a>. When used in the <code><a>NFCWriter.push()</a></code>
+      method, its <dfn>NDEFMessageInit.url</dfn> member represents a <a>URL path</a>
+      used for constructing the <a>Web NFC Id</a> of the pushed <a>Web NFC content</a>.
+    </p>
   </section>
 
-  <section> <h3>The <dfn>NDEFRecord</dfn> dictionary</h3>
+  <section> <h3>The <dfn>NDEFRecord</dfn> interface</h3>
     <p>
       The content of any <a>NDEF record</a> is exposed by the
-      <a>NDEFRecord</a> dictionary:
+      <a>NDEFRecord</a> interface:
     </p>
     <pre class="idl">
       typedef (DOMString or unrestricted double or ArrayBuffer or Dictionary) NDEFRecordData;
 
-      dictionary NDEFRecord {
+      [Constructor(NDEFRecordInit recordInit), Exposed=Window]
+      interface NDEFRecord {
+        readonly attribute NDEFRecordType recordType;
+        readonly attribute USVString mediaType;
+        NDEFRecordData data();
+      };
+
+      dictionary NDEFRecordInit {
         NDEFRecordType recordType;
         USVString mediaType;
         NDEFRecordData data;
       };
     </pre>
     <p>
       The <dfn>NDEFRecordData</dfn> is a union type representing data types allowed
```

Comment:
Shouldn't you write that in Web IDL then?

Reply:
Eh... Sorry, I didn't get your point.  `NDEFRecordData` is already defined in [here](https://github.com/w3c/web-nfc/pull/240/files#diff-eacf331f0ffc35d4b482f1d15a887d3bR1186).